components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/AndroidDownloadManager.kt

```diff
@@ -99,19 +111,15 @@ class AndroidDownloadManager(
      * broadcast receiver if there are no more queued downloads.
      */
     override fun onReceive(context: Context, intent: Intent) {
-        if (queuedDownloads.isEmpty()) unregisterListeners()
```

Comment:
We must remove these so future `onDownloadCompleted` calls get triggered (even if the user just does a `tryAgain`. This actually fixes an existing bug: Download fails, press "try again" in notification, `onDownloadCompleted` does not get called. 

The listeners are still unregistered in `onStop` so there's no leak here.